tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/Log4Net/DirectSubmission/AppenderCollectionLegacyIntegration.cs

```diff
@@ -0,0 +1,62 @@
+﻿// <copyright file="AppenderCollectionLegacyIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Logging.DirectSubmission;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.Log4Net.DirectSubmission
+{
+    /// <summary>
+    /// AppenderCollection.ToArray() calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "log4net",
+        TypeName = "log4net.Appender.AppenderCollection",
+        MethodName = "ToArray",
+        ReturnTypeName = "log4net.Appender.IAppender[]",
+        ParameterTypeNames = new string[0],
+        MinimumVersion = "1.0.0",
+        MaximumVersion = "1.*.*",
+        IntegrationName = nameof(IntegrationId.Log4Net))]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class AppenderCollectionLegacyIntegration
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<AppenderCollectionLegacyIntegration>();
+        private static bool _logWritten = false;
```

Comment:
I must be missing something obvious but I don't see any difference with the `AppenderCollectionIntegration`. Can't we mutualize the code?

Reply:
The difference is that `AppenderCollectionLegacyIntegration` adds a `DirectSubmissionLog4NetLegacyAppender`, while `AppenderCollectionIntegration` adds a `DirectSubmissionLog4NetAppender`

Reply:
Thanks. I had missed it